src/lib/Microsoft.Health.Fhir.Ingest/Service/EventProcessingMeter.cs

```diff
@@ -0,0 +1,35 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.EventHubs;
+
+namespace Microsoft.Health.Fhir.Ingest.Service
+{
+    public class EventProcessingMeter : IEventProcessingMeter
+    {
+        public Task<EventStats> CalculateEventStats(EventData[] events)
+        {
+            double ingressSizeBytes = 0;
+
+            foreach (var e in events)
+            {
+                var bodySizeBytes = e.Body.Array.Length;
+                ingressSizeBytes = ingressSizeBytes + bodySizeBytes;
+                ingressSizeBytes = e.Properties.Aggregate(ingressSizeBytes, (current, entry) => current + Encoding.UTF8.GetBytes(entry.Key + entry.Value).Length);
```

Comment:
>entry.Key + entry.Value [](start = 129, length = 23)

.NET strings are immutable, this will cause a new string to be created in memory.  Would be better to do GetBytes on each string section and add.  Also suggest using Encoding.UTF8.GetByteCount() over GetBytes.Length.

Reply:
That makes sense. Updated. Agreed GetByteCount is better